### PR TITLE
Fix close issue but time watcher still running

### LINKED ISSUE #11374: The timetracker will not be stop from a issue reference in the message of a PR
- Gitea version (or commit ref): 99082eebd76019fc608e44c4464da6398f51dccd = ~ v1.12.0 
- Git version: 2.20.1
- Operating system: Linux Debian Buster
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (https://try.gitea.io/luwol03/test/issues/1)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description
the time tracker will be stoped, if I refer a issue in a pr commit message. But if I close a issue via refer it in a pr message it will not be closed!

## Screenshots
image

### PR DESCRIPTION
**Pull request by lunny**
Fix #11374

### CODE REVIEW

File: services/issue/status.go [hunk 1]
```diff
@@ -10,12 +10,31 @@ import (
...
-		return
+		// Don't return an error when dependencies are open as this would let the push fail
+		if models.IsErrDependenciesLeft(err) {
+			return stopTimerIfAvailable(doer, issue)
+		}
+		return err
 	}
 
-	notification.NotifyIssueChangeStatus(doer, issue, comment, isClosed)
+	if err := stopTimerIfAvailable(doer, issue); err != nil {
```
Comment (wxiaoguang) [inline services/issue/status.go]: Should we check `closed` here?

Comment (lunny) [reply services/issue/status.go]: done.

**Review by wxiaoguang — APPROVED** [review]

File: models/issue_stopwatch.go [hunk 2]
```diff
@@ -13,6 +13,26 @@ import (
...
 )
 
+// ErrIssueStopwatchNotExist represents an error that stopwatch is not exist
+type ErrIssueStopwatchNotExist struct {
+	UserID  int64
+	IssueID int64
+}
+
+func (err ErrIssueStopwatchNotExist) Error() string {
+	return fmt.Sprintf("issue stopwatch is not exist[uid: %d, issue_id: %d", err.UserID, err.IssueID)
```
Comment (Gusted) [inline models/issue_stopwatch.go]:
```suggestion
	return fmt.Sprintf("issue stopwatch doesn't exist [uid: %d, issue_id: %d", err.UserID, err.IssueID)
```

File: models/issue_stopwatch.go [hunk 3]
```diff
@@ -13,6 +13,26 @@ import (
...
+}
+
+// ErrIssueStopwatchAlreadyExist represents an error that stopwatch is already exist
+type ErrIssueStopwatchAlreadyExist struct {
+	UserID  int64
+	IssueID int64
+}
+
+func (err ErrIssueStopwatchAlreadyExist) Error() string {
+	return fmt.Sprintf("issue stopwatch is already exist[uid: %d, issue_id: %d", err.UserID, err.IssueID)
```
Comment (Gusted) [inline models/issue_stopwatch.go]:
```suggestion
	return fmt.Sprintf("issue stopwatch already exists [uid: %d, issue_id: %d", err.UserID, err.IssueID)
```

File: models/issue_stopwatch.go (hunk 2 above)
Comment (wxiaoguang) [reply models/issue_stopwatch.go]: `doesn't exist` ?

File: models/issue_stopwatch.go (hunk 3 above)
Comment (wxiaoguang) [reply models/issue_stopwatch.go:33]: `exist` => `exists` ?

File: models/issue_stopwatch.go (hunk 2 above)
Comment (Gusted) [reply models/issue_stopwatch.go]: Great, my english has once again failed me 😄

**Review by Gusted — APPROVED** [review]

**Comment by zeripath** [comment]
I think this is incorrect and rather the error handling should be happening at the caller end.

**Pull request merged**